azure/modules/ad_application/main.tf

```diff
@@ -98,7 +98,7 @@ resource "azurerm_role_assignment" "grant_reader_role_to_subscriptions" {
   count = var.create ? length(data.azurerm_subscriptions.available.subscriptions) : 0
   scope = "/subscriptions/${data.azurerm_subscriptions.available.subscriptions[count.index].subscription_id}"
 
-  principal_id         = azuread_service_principal.lacework[0].id
+  principal_id         = local.service_principal_id
```

Comment:
This is just using the service principal id defined above in the `locals {}` block.